projects/packages/identity-crisis/src/class-rest-endpoints.php

```diff
@@ -233,4 +276,20 @@ public static function url_secret_permission_check() {
 				array( 'status' => rest_authorization_required_code() )
 			);
 	}
+
+	/**
+	 * The endpoint is only available on non-connected sites.
+	 * use `/identity-crisis/url-secret` for connected sites.
+	 *
+	 * @return true|WP_Error
+	 */
+	public static function compare_url_secret_permission_check() {
+		return ( new Connection_Manager() )->is_connected()
+			? new WP_Error(
```

Comment:
Another drive-by comment: The error code and message here can be misleading, since this is not the result of a permissions' check. Maybe we could replace it with a more relevant `WP_Error`?

Reply:
Good point, fixed 👍